pool/allocator.go

```diff
@@ -38,6 +40,9 @@ func NewPoolAllocator(ctx context.Context, timeout time.Duration, factory Factor
 			return nil, err
 		}
 
+		// should be in factory.SpawnWorkerWithTimeout, but this method is missing cfg parameter
+		w.SetMaxExecs(calculateMaxExecs(cfg.MaxJobs, cfg.MaxJobsDispersion))
```

Comment:
I'm not sure about PHP-like getters/setter. Usually, in Go, this is a bad practice. It's better to use Options (like in the `InitBaseWorker`). There are a few setters currently exists, but I think, that the approach should be rethought to eliminate them.